compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -1737,11 +1740,13 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         // 2. Produce something like `&'0 i32 : Copy`
         // 3. Re-bind the regions back to `for<'a> &'a i32 : Copy`
 
+        // Vec isn't Copy
+        let binders = types.rebind(());
```

Comment:
can you instead use a newtype and a method on `Binder` here. Something `types.rebind_token` or something like that?

Reply:
So you mean, like, have something like `struct BoundsToken;` then on `Binder` have `fn bound_vars(&self) -> Binder<BoundsToken>` and `fn rebind_token<T>(vars: Binder<BoundsToken>, value: T) -> Binder<T>`?

I could, but that seems like quite a lot of boilerplate for a not-common case. Even more so: once #76814 lands that will essentially be unnecessary (which would increase the diff on that PR).

Reply:
My idea for #76814 would be `fn Binder::bound_vars(&self) -> &'tcx List<BoundVariableKind>` and `fn List<BoundVariableKind>::bind` or `fn Binder::bind_with(&'tcx List<BoundVariableKind>, T)`.

And until #76814 lands we use a dummy `struct BoundVariables;` for that

Reply:
Ah okay. That works for me. Previously, I know there were some concerns about that approach; but now, it's more clear that tracking bound vars is doable (given #76814), so I think it's fine now.